README.md

```diff
@@ -2,58 +2,162 @@
 
 ## Introduction
 
-This plugin is meant to make it easy to quickstart development of new OBS plugins. It includes:
+The plugin template is meant to be used as a starting point for OBS Studio plugin development. It includes:
+
+* Boilerplate plugin source code
+* A CMake project file
+* GitHub Actions workflows and repository actions
+
+## Set Up
+
+The plugin project is set up using the included `buildspec.json` file - the following fields should be customized for an actual plugin:
+
+* `name`: The plugin name
+* `version`: The plugin version
+* `author`: Actual name or nickname of the plugin's author
+* `website`: URL of a website associated with the plugin
+* `email`: Contact email address associated with the plugin
+* `uuids`
+    * `macosPackage`: Unique (**!**) identifier for the macOS plugin plugin
+    * `macosInstaller`: Unique (**!**) identifier for the macOS plugin installer
+    * `windowsApp`: Unique (**!**) identifier for the Windows plugin installer
+
+These values are read and processed automatically by the CMake build scripts, so no further adjustments in other files are needed.
+
+### Platform Configuration
+
+Platform-specific settings are set up in the `platformConfig` section of the buildspec file:
+
+* `bundleId`: macOS bundle identifier for the plugin. Should be unique and follow a reverse-URL scheme.
+
+### Set Up Build Dependencies
+
+Just like OBS Studio itself, plugins need to be built using dependencies available either via the `obs-deps` repository (Windows and macOS) or via a distribution's package system (Linux).
+
+#### Windows and macOS
+
+Windows and macOS dependency downloads are configured in the `buildspec.json` file:
+
+* `dependencies`:
+    * `obs-studio`: Version of OBS Studio to build plugin with - needed for `libobs` and `obs-frontend-api` (see below)
+    * `prebuilt`: Prebuilt OBS Studio dependencies
+    * `qt6`: Prebuilt version of Qt6 as used by OBS Studio
+* `tools`: Contains additional build tools used by CI
+
+The values should be kept in sync with OBS Studio releases and the `buildspec.json` file in use by the main project to ensure that the plugin is developed and built in sync with its target environment.
```

Comment:
I don't think this requires specific mention - API compatibility is a concern of `obs-studio` and breaking changes would need to be mentioned upon updates there.

Also the plugintemplate is only supposed to be a starting point for plugin development, there is not (and should not) be the idea that plugin developers should rebase their code on this template. 

So it's entirely up to plugin developers to change/adapt this value and its _their_ task to figure out API compatibility and target their plugin accordingly.

Reply:
Again, I won't block on the mention (or lack thereof) of the caveat in documentation at this time. I just felt it warranted mentioning here. Plugins compiled for older OBS versions _generally_ work fine in newer OBS versions, but plugins compiled for the latest OBS will not load in older versions of OBS if the libobs structs have changed between those versions. It was mostly a point that compiling for the latest version of OBS will lock out users of older OBS versions. While _we_ do not support older versions of OBS, we won't mandate that plugin devs do the same.

> there is not (and should not) be the idea that plugin developers should rebase their code on this template.

Does that mean we should close and reject any Issue submissions that concern older revisions of the plugin template? While it was clear to me that we would not force plugin developers to update/rebase their repositories to this revision, I thought it would be encouraged, as hypothetically most of the files changed are ones they should not have had to modify.

> So it's entirely up to plugin developers to change/adapt this value and its their task to figure out API compatibility and target their plugin accordingly.

Well, yes, that much I agree with. I just wanted something, in the future perhaps, to clarify the ramifications.

Reply:
> Again, I won't block on the mention (or lack thereof) of the caveat in documentation at this time. I just felt it warranted mentioning here. Plugins compiled for older OBS versions _generally_ work fine in newer OBS versions, but plugins compiled for the latest OBS will not load in older versions of OBS if the libobs structs have changed between those versions. It was mostly a point that compiling for the latest version of OBS will lock out users of older OBS versions. While _we_ do not support older versions of OBS, we won't mandate that plugin devs do the same.

Got it - adding a short paragraph about "Which Version of OBS Studio To Build With" should address this.

> > there is not (and should not) be the idea that plugin developers should rebase their code on this template.
> 
> Does that mean we should close and reject any Issue submissions that concern older revisions of the plugin template? While it was clear to me that we would not force plugin developers to update/rebase their repositories to this revision, I thought it would be encouraged, as hypothetically most of the files changed are ones they should not have had to modify.

It was my expectation that the template just serves as that: A template. A plugin developer forks it, but then adjusts any and all parts as needed, removing unnecessary code (i.e., Qt6 selection if no UI code is created), removing unnecessary CI stuff (i.e., macOS and Linux build jobs if the plugin only targets Windows systems) and making it "their own".

The problem with just rebasing their local forks is that we can never guarantee that template changes will work with the codebase of an actual in-development plugin. Authors are free to update their own forks that way, but that obviously requires them to know what each part does and not just blindly merge changes.

But what we can't guarantee (and we shouldn't suggest) is to say "just update the C/C++ code, leave everything else as-is and update from upstream". I'd rather say "fork the template, familiarise yourself with all parts of it, customise it to your needs". 

Ideally they should never have to take another look at the template again after - even when it comes how to build `libobs` or pull in dependencies, plugin authors are free to change all of it (because either way, they are responsible for making their customisations work).

Reply:
Discussed off-thread.